include-blang/cooperative_groups.h

```diff
@@ -0,0 +1,32 @@
+#ifndef _COOPERATIVE_GROUPS_H_
+#define _COOPERATIVE_GROUPS_H_
+
+namespace cooperative_groups {
+  class grid_group {
+    public:
+      __forceinline__ __device__ void sync() const {
+          bugle_grid_barrier();
+      }
+  };
+  
+  class thread_block {
+    public:
+      __forceinline__ __device__ void sync() const {
+          bugle_barrier(true, true);
+      }
+  };
+  
+  __forceinline__ __device__ grid_group this_grid();
```

Comment:
Apparently this call and the next have some barrier-like properties.

>  In the documentation I read: "Although you can create an implicit group anywhere in the code, it is dangerous to do so. Creating a handle for an implicit group is a collective operation - all threads in the group must participate. If the group was created in a conditional branch that not all threads reach, this can lead to deadlocks or data corruption.

We could checks that, but this is out-of-scope of this PR. I'll make sure to file an issue for that.